academy/ibc/channels.md

```diff
@@ -0,0 +1,166 @@
+
+```go
+// ChannelOpenInit defines a rpc handler method for MsgChannelOpenInit.
+// ChannelOpenInit will perform 04-channel checks, route to the application
+// callback, and write an OpenInit channel into state upon successful execution.
+func (k Keeper) ChannelOpenInit(goCtx context.Context, msg *channeltypes.MsgChannelOpenInit) (*channeltypes.MsgChannelOpenInitResponse, error) {
+  ctx := sdk.UnwrapSDKContext(goCtx)
+
+  // Lookup module by port capability
+  module, portCap, err := k.PortKeeper.LookupModuleByPort(ctx, msg.PortId)
+  if err != nil {
+    return nil, sdkerrors.Wrap(err, "could not retrieve module from port-id")
+  }
+
+  ...
+
+  // Perform 04-channel verification
+  channelID, cap, err := k.ChannelKeeper.ChanOpenInit(
+    ctx, msg.Channel.Ordering, msg.Channel.ConnectionHops, msg.PortId,
+    portCap, msg.Channel.Counterparty, msg.Channel.Version,
+  )
+
+  ...
+
+  // Perform application logic callback
+  if err = cbs.OnChanOpenInit(ctx, msg.Channel.Ordering, msg.Channel.ConnectionHops, msg.PortId, channelID, cap, msg.Channel.Counterparty, msg.Channel.Version); err != nil {
+    return nil, sdkerrors.Wrap(err, "channel open init callback failed")
+  }
+
+  // Write channel into state
+  k.ChannelKeeper.WriteOpenInitChannel(ctx, msg.PortId, channelID, msg.Channel.Ordering, msg.Channel.ConnectionHops, msg.Channel.Counterparty, msg.Channel.Version)
+
+  return &channeltypes.MsgChannelOpenInitResponse{
+    ChannelId: channelID,
+  }, nil
+}
+```
+
+<HighlightBox type="info">
+
+**Capabilities**
+
+IBC is intended to work in execution environments where modules do not necessarily trust each other. This security is accomplished using a [dynamic
+capability store](https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-003-dynamic-capability-store.md. This binding strategy prevents other modules from using the particular port or channel since those modules do not own the appropriate capability. While this background information is useful, IBC application developers should not need to modify this lower level abstraction, other than setting the capabilities appropriately in `app.go`.
+
+</HighlightBox>
+
+## Application Packet Flow
+
+As stated above, application modules communicate with each other by sending packets over IBC channels. However, IBC modules do not directly pass these messages to each other over the network. Rather, the module will commit some state reflecting the transaction execution to a precisely defined path reserved for a specific message type and a specific counterparty. For example, as part of an ICS20 token transfer, the bank module would escrow the portion of tokens to be transferred and store
```

Comment:
Please note that the final sentence in line 104 ends incomplete. Possibly this could be a very simple fix:

_For example, as part of an ICS20 token transfer, the bank module would escrow the portion of tokens to be transferred and store**d.**_

However, the sentence could continue at length from this point (_"...transferred and store**d in an X Y Z**"_). Please review and confirm.